Tests/FluentAssertions.Specs/Collections/CollectionAssertionSpecs.BeInAscendingOrder.cs

```diff
@@ -153,6 +153,71 @@ public void When_asserting_single_element_collection_by_property_expression_orde
             act.Should().NotThrow();
         }
 
+        [Fact]
+        public void When_asserting_single_element_collection_by_converted_property_expression_ordered_in_ascending_it_should_succeed()
+        {
+            // Arrange
+            var collection = new SomeClass[]
+            {
+                new() { Text = "a", Number = 1 }
+            };
+
+            // Act
+            Action act = () => collection.Should().BeInAscendingOrder(o => (float)o.Number);
+
+            // Assert
+            act.Should().NotThrow();
+        }
+
+        [Fact]
+        public void When_asserting_single_element_collection_by_indexed_List_property_expression_ordered_in_ascending_it_should_succeed()
+        {
+            // Arrange
+            var collection = new[]
+            {
+                new List<SomeClass> { new() { Text = "a", Number = 1 } }
+            };
+
+            // Act
+            Action act = () => collection.Should().BeInAscendingOrder(o => o[0].Number);
+
+            // Assert
+            act.Should().NotThrow();
+        }
+
+        [Fact]
+        public void When_asserting_single_element_collection_by_indexed_Array_property_expression_ordered_in_ascending_it_should_succeed()
```

Comment:
❓ Is this scenario really covering a different path compared to the previous test case?

Reply:
Yes, there's a difference between an array indexer and a List indexer.
See the currently uncovered `Convert`, `ArrayIndex` and `Call` cases

https://coveralls.io/builds/58517312/source?filename=Src%2FFluentAssertions%2FCommon%2FExpressionExtensions.cs

Reply:
I should have checked that myself. I can always count on the thoroughness of @jnyrup 😜